airbyte-cdk/python/unit_tests/test/mock_http/test_mocker.py

```diff
@@ -10,51 +10,134 @@
+            [HttpResponse(_A_RESPONSE_BODY, 1), HttpResponse(_ANOTHER_RESPONSE_BODY, 2)],
+        )
+
+        first_response = requests.post(_A_URL, params=_SOME_QUERY_PARAMS, headers=_SOME_HEADERS, data=_SOME_REQUEST_BODY_STR)
+        second_response = requests.post(_A_URL, params=_SOME_QUERY_PARAMS, headers=_SOME_HEADERS, data=_SOME_REQUEST_BODY_STR)
+
+        assert first_response.text == _A_RESPONSE_BODY
         assert first_response.status_code == 1
-        assert second_response.text == _ANOTHER_BODY
+        assert second_response.text == _ANOTHER_RESPONSE_BODY
         assert second_response.status_code == 2
 
+    @HttpMocker()
+    def test_given_body_is_mapping_pass_as_json_when_decorate_then_match_query(self, http_mocker):
+        http_mocker.post(
+            HttpRequest(_A_URL, _SOME_QUERY_PARAMS, _SOME_HEADERS, _SOME_REQUEST_BODY_MAPPING),
+            _A_RESPONSE,
+        )
+
+        requests.post(_A_URL, params=_SOME_QUERY_PARAMS, headers=_SOME_HEADERS, json=_SOME_REQUEST_BODY_MAPPING)
+
+    @HttpMocker()
+    def test_given_body_is_str_pass_as_json_when_decorate_then_match_query(self, http_mocker):
+        http_mocker.post(
+            HttpRequest(_A_URL, _SOME_QUERY_PARAMS, _SOME_HEADERS, _SOME_REQUEST_BODY_STR),
+            _A_RESPONSE,
+        )
+
+        requests.post(_A_URL, params=_SOME_QUERY_PARAMS, headers=_SOME_HEADERS, json=_SOME_REQUEST_BODY_STR)
+
+    @HttpMocker()
+    def test_given_body_is_str_pass_as_data_when_decorate_then_match_query(self, http_mocker):
+        http_mocker.post(
+            HttpRequest(_A_URL, _SOME_QUERY_PARAMS, _SOME_HEADERS, _SOME_REQUEST_BODY_STR),
+            _A_RESPONSE,
+        )
+
+        requests.post(_A_URL, params=_SOME_QUERY_PARAMS, headers=_SOME_HEADERS, data=_SOME_REQUEST_BODY_STR)
+
+    @HttpMocker()
+    def test_given_body_is_json_str_with_formatting_when_decorate_then_match_query(self, http_mocker):
+        http_mocker.get(
+            HttpRequest(_A_URL, _SOME_QUERY_PARAMS, _SOME_HEADERS, {"a body": "body"}),
+            _A_RESPONSE,
+        )
+
+        requests.get(_A_URL, params=_SOME_QUERY_PARAMS, headers=_SOME_HEADERS, data='{\n  "a body": "body"\n}')
+
+    @HttpMocker()
+    def test_given_body_is_mapping_pass_as_json_different_order_when_decorate_then_match_query(self, http_mocker):
+        http_mocker.post(
+            HttpRequest(_A_URL, _SOME_QUERY_PARAMS, _SOME_HEADERS, {"first_field": "first_value", "second_field": 2}),
+            _A_RESPONSE,
+        )
+
+        requests.post(_A_URL, params=_SOME_QUERY_PARAMS, headers=_SOME_HEADERS, json={"second_field": 2, "first_field": "first_value"})
+
+    @HttpMocker()
+    def test_given_body_is_json_str_different_order_when_decorate_then_match_query(self, http_mocker):
+        http_mocker.post(
+            HttpRequest(_A_URL, _SOME_QUERY_PARAMS, _SOME_HEADERS, '{"first_field": "first_value", "second_field": 2}'),
```

Comment:
If as tester I set this as a string, shouldn't I expect the formatting to match? I fear that this means we loose the ability to match on a specific string right?

Reply:
If it is not a JSON string, then the formatting will be kept, otherwise both JSON strings will have the same formatting when they're compared. Does it make sense?

Reply:
If people would respect specs, I would agree with this. However, there are numerous cases on the internet where people just do shenanigans ([example](https://stackoverflow.com/q/4515676/2800559)). If we have a source that does not respect the JSON spec and we want to validate that, now we can't which feels like it's a gap in the solution we propose for integration tests. Does that make sense?

Reply:
As discussed, we will check without applying formatting as soon as one or the other requests has a Mapping